mmengine/hooks/ema_hook.py

```diff
@@ -180,13 +193,16 @@ def after_load_checkpoint(self, runner, checkpoint: dict) -> None:
 
         # Support load checkpoint without ema state dict.
         else:
-            print_log(
-                'There is no `ema_state_dict` in checkpoint. '
-                '`EMAHook` will make a copy of `state_dict` as the '
-                'initial `ema_state_dict`', 'current', logging.WARNING)
-            self.ema_model.module.load_state_dict(
+            if self.resume:
+                print_log(
+                    'There is no `ema_state_dict` in checkpoint. '
+                    '`EMAHook` will make a copy of `state_dict` as the '
+                    'initial `ema_state_dict`', 'current', logging.WARNING)
+            _load_checkpoint_to_model(
+                self.ema_model.module,
                 copy.deepcopy(checkpoint['state_dict']),
-                strict=self.strict_load)
+                strict=self.strict_load,
+                revise_keys=self.revise_keys)
```

Comment:
Seems to be unnecessary because the keys in `state_dict` have no "module." prefix.